### PR TITLE
Added tree_diameter.cpp

### PR DESCRIPTION
**Pull request by rakki-18**
#### Description of Change

Added tree_diameter.cpp that finds the diameter of a tree.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1227"></a>

### CODE REVIEW

**Comment by rakki-18** [comment]
@kvedala Could you please review this ?

File: graph/tree_diameter.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * [Tree Diameter](https://en.wikipedia.org/wiki/Treewidth)
+ * 
+ * @author [Rakshit P](https://github.com/rakki-18)
+ */
+ 
+#include<algorithm>
+#include<iostream>
+#include<vector>
+
+std::vector<std::vector<int>>graph;
```
Comment (fhlasek) [inline graph/tree_diameter.cpp:15]: Please the `using` declaration to define the common graph structure:
https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/graph/breadth_first_search.cpp#L66

File: graph/tree_diameter.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+int tree_diameter;
```
Comment (fhlasek) [inline graph/tree_diameter.cpp:16]: Don't create global variables, define a function which takes all the input values as parameters and returns the output as return value.

File: graph/tree_diameter.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+int number_of_vertices, number_of_edges;
+int longest_path_from_node(int u, int parent)
```
Comment (fhlasek) [inline graph/tree_diameter.cpp:18]: Please document the function's interface.

**Comment by rakki-18** [comment]
@fhlasek thanks for reviewing! I will make the changes and then submit a pr

**Comment by kvedala** [comment]
> @fhlasek thanks for reviewing! I will make the changes and then submit a pr

Do not open a new Pr. Make changes to this current PR. This way, the reviewers can track the changes.

**Pull request closed without merging**